### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by Tracy-miranja**
- [ ] Add a column species of type string to your animals table. Modify your schema.sql file.
- [ ] Insert the following data:
- [ ] Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
- [ ] Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- [ ] Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
- [ ] Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- [ ] Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- [ ] Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- [ ] Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- [ ] Inside a transaction update the animals' table by setting the species column to unspecified.Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- [ ] Inside a transaction:
- [ ] Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- [ ] Verify that changes were made.
- [ ] Commit the transaction.
- [ ] Verify that changes persist after commit.
- [ ] delete all records in the animals table, then roll back the transaction.
- [ ] verify if all records in the animals table still exists.
- [ ] Inside a transaction:
- [ ] Delete all animals born after Jan 1st, 2022.
- [ ] Create a savepoint for the transaction.
- [ ] Update all animals' weight to be their weight multiplied by -1.
- [ ] Rollback to the savepoint
- [ ] Update all animals' weights that are negative to be their weight multiplied by -1.
- [ ] Commit transaction.
- [ ] Write queries to answer the following questions:
- [ ] How many animals are there?
- [ ] How many animals have never tried to escape?
- [ ] What is the average weight of animals?
- [ ] Who escapes the most, neutered or not neutered animals?
- [ ] What is the minimum and maximum weight of each type of animal?
- [ ] What is the average number of escape attempts per animal type of those
trasaction 2
rollback

QUERY1
species22
 born between 1990 and 2000?
- [ ]

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -5,4 +5,18 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE neutered_weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE neutered_weight_kg >= 10.4 AND neutered_weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE neutered_weight_kg >= 10.4 AND neutered_weight_kg <= 17.3;
```
Comment (Zuhaib042) [inline queries.sql:8]: - [ ] I dont see the transactions os these requirements in this queries.sql file. also some of the screenshots of these transactions also missing in your PR. kindly add the following transactions here in this file and also carefully add the screenshots of these transactions in order. You may double check if your PR has the screenshots of all the transactions and queries.
required transactions:
Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Verify that changes were made.
- Commit the transaction.
- Verify that changes persist after commit.
- Take a screenshot of the results of your actions.
- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
- Take a screenshot of the results of your actions. 
- Remember to add all these transactions to your queries.sql file.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @Tracy-miranja 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Well written queries
✅ Descriptive PR 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Tracy-miranja) [reply queries.sql:8]: Thanks for the feedback @Zuhaib042

File: queries.sql
Comment (OyePriscilla) [inline queries.sql:1]: - [ ] Kindly implement this omitted requirement as part of the query as well. ℹ️ 

Untitled

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,27 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+UPDATE animals SET species = 'digimon' where name like '%mon';
+UPDATE animals SET species = 'pokemon' where species IS NULL OR species = '';
+SELECT name, species FROM animals;
+COMMIT;
+SELECT name, species FROM animals;
+BEGIN; 
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
```
Comment (OyePriscilla) [inline queries.sql:23]: - [optional] Please kindly check if the data was deleted as expected after the delete command. 👍 
```suggestion
BEGIN; 
DELETE FROM animals;
SELECT * FROM animals;
ROLLBACK;
SELECT * FROM animals;
```

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Tracy-miranja 👏 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use database transactions ✔️ 
Modify and delete data in SQL ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql
Comment (Tracy-miranja) [reply queries.sql:1]: Can you please be clear, I don't get what you are trying to point out.@OyePriscilla

**Review by roshan-bajgain — APPROVED** [review]
Hi @Tracy-miranja, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- GitHub flow is followed. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**